### ISSUE TITLE
Allow arrays allocated with malloc() to be expanded in Watch window

### ISSUE THREAD

**Issue by kevinmoran** [microsoft/vscode-cpptools#277]
Visual Studio has the syntax "_ptr_to_array, 10_" to show first 10 elements of a malloc-ed array in the watch window. This would be an extremely nice feature, inspecting dynamic memory is quite an irritating process in VsCode at the moment. Thanks.

**Comment by jacdavis** [comment]
If you are using the windows debugger with vscode (not gdb, but the real windows debugger) this syntax is supported in the watch window. If you are using gdb or lldb, you'd need to use the syntax as supported by the expression evaluator for the debugger being used. Assuming you are using gdb, this documentation shows you how:
http://www.delorie.com/gnu/docs/gdb/gdb_54.html

Note that you'd need to do this from the debugger console with -exec in front of the command.

**Comment by kevinmoran** [comment]
I'm still having trouble with this, if anyone can point out what I'm doing wrong I'd appreciate it.

I allocated an array using the following:
    int *array = (int *) malloc (32 * sizeof (int));

I set a breakpoint after this line and started a debug session using GDB. In the Debug Console I can print the first element of the array using the command
```
    -exec p *array
```

which gives the output:
```
    (int) $0 = 0
```
I try to print the entire array using the standard GDB syntax:
```
    -exec p *array@32
```

But this gives no output whatsoever. I've tried and searched everywhere, can anyone tell me what I'm doing wrong?
